src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs

```diff
@@ -253,6 +253,28 @@ public override void ExecuteSiteRecoveryCmdlet()
                             vmNic.EnableAcceleratedNetworkingOnTfo ?? false;
                     }
 
+                    List<PSIPConfigInputDetails> ipConfigList = null;
+                    if (this.IPConfig == null || this.IPConfig.ToList().Count == 0)
+                    {
+                        ipConfigList = vmNic.IpConfigs?.Select(ip => ConvertToPSIPConfig(ip)).ToList() ?? null;
```

Comment:
?.ToList()

Reply:
vmNic.IPConfigs is of type IList<> while IPConfigs in ASRVMNicConfig is of type List<>. So

Reply:
Null check.

Reply:
done